### ISSUE TITLE
Allow defining directives inline, ie. after a command

### ISSUE THREAD

**Issue by setpill** [koalaman/shellcheck#1840]
This is useful especially in Dockerfiles (ie. when using shellcheck through hadolint). In Dockerfiles you cannot have true multiline shell scripts without really ugly hacks, so having to put the directive *before* the affected command is impossible as shell scripts do not have comments that end before the line ends.

#### For new checks and feature suggestions
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```Dockerfile

RUN cp $(ls | grep "something" | sort | uniq) somewhere # shellcheck disable=SC2046

```

#### Here's what shellcheck currently says:

```
SC2046 Quote this to prevent word splitting.
```

#### Here's what I wanted or expected to see:

Nothing

**Comment by brother** [comment]
Maybe you shouldn't pass dockerfile to shellcheck? After all it's not a supported format.

**Comment by setpill** [comment]
I don't think that's what hadolint does under the hood. I haven't looked at the code, but I would assume it extracts the inline shell code from the Dockerfile and passes that to shellcheck.

**Comment by brother** [comment]
Assumptions. Indeed. Ok.

**Comment by setpill** [comment]
Being cognizant of assumptions is important. They are useful time management tools, but they have their limitations. This is not really the forum for such discussion though.
